### PR TITLE
Use eclipse.jdt.ls 1.16.0-SNAPSHOT in TP configurator

### LINKED ISSUE #291: Build failure due to missing eclipse.jdt.ls target file
When I try to build/test the latest commit, I get an error saying:

```
java.lang.RuntimeException: Invalid syntax in target definition https://repo.eclipse.org/content/repositories/jdtls-snapshots/org/eclipse/jdt/ls/org.eclipse.jdt.ls.tp/1.16.0-SNAPSHOT/org.eclipse.jdt.ls.tp-1.16.0-20220915.034113-10.target: I/O error while reading target definition file: https://repo.eclipse.org/content/repositories/jdtls-snapshots/org/eclipse/jdt/ls/org.eclipse.jdt.ls.tp/1.16.0-SNAPSHOT/org.eclipse.jdt.ls.tp-1.16.0-20220915.034113-10.target
```

It looks like the target file we referenced doesn't exist anymore; when I go to the url listed in the logs, I get a 404. I think this is because snapshot builds are eventually deleted.

### PR DESCRIPTION
**Pull request by datho7561**
Closes #291

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

**Comment by rgrunber** [comment]
This won't work because the stable build of JDT-LS 1.15.0 refers to the 4.25-I-builds update site, which is gone now that 4.25 was released.

**Comment by rgrunber** [comment]
@datho7561 , can you confirm for me if the following change works. This is purely applied to master, with nothing else (testing on microprofile.jdt only) :

```diff
diff --git a/microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target b/microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target
index a5ed56c..f41360a 100644
--- a/microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target
+++ b/microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target
@@ -2,7 +2,6 @@
 <?pde version="3.8"?>
 <target name="LSP4MP Target Platform">
        <locations>
-               <location type="Target" uri="https://repo.eclipse.org/content/repositories/jdtls-snapshots/org/eclipse/jdt/ls/org.eclipse.jdt.ls.tp/1.16.0-SNAPSHOT/org.eclipse.jdt.ls.tp-1.16.0-20220915.034113-10.target"/>
                <location includeDependencyDepth="infinite" includeDependencyScopes="compile" includeSource="true" missingManifest="generate" type="Maven">
                        <dependencies>
                                <dependency>
diff --git a/microprofile.jdt/pom.xml b/microprofile.jdt/pom.xml
index 67b73f8..0a33c73 100644
--- a/microprofile.jdt/pom.xml
+++ b/microprofile.jdt/pom.xml
@@ -59,6 +59,11 @@
               <artifactId>org.eclipse.lsp4mp.jdt.tp</artifactId>
               <version>${project.version}</version>
             </artifact>
+            <artifact>
+              <groupId>org.eclipse.jdt.ls</groupId>
+              <artifactId>org.eclipse.jdt.ls.tp</artifactId>
+              <version>1.16.0-SNAPSHOT</version>
+            </artifact>
           </target>
         </configuration>
       </plugin>
```

**Comment by datho7561** [comment]
> can you confirm for me if the following change works

Doesn't look like it. Eclipse doesn't like this, and it doesn't seem to work for the CI either

File: microprofile.jdt/pom.xml [hunk 1]
```diff
@@ -38,6 +39,10 @@
       <layout>p2</layout>
       <url>https://download.eclipse.org/jdtls/milestones/1.15.0/repository/</url>
     </repository>
+    <repository>
+      <id>jdt.ls.maven</id>
+      <url>https://repo.eclipse.org/content/repositories/jdtls-releases/</url>
```
Comment (rgrunber) [inline microprofile.jdt/pom.xml:44]: Changes this to `jdt-snapshots` . Releases wouldn't have 1.16.0-SNAPSHOT.

Comment (rgrunber) [reply microprofile.jdt/pom.xml:44]: Also, I to get it to work locally, I needed to change the `id` to something like `jdt.ls.maven2` :\ which seems odd, but it does work afterwards.

Comment (datho7561) [reply microprofile.jdt/pom.xml]: Looks like it works for me! Didn't even need to change the id

Comment (datho7561) [reply microprofile.jdt/pom.xml]: Thanks for the help!

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target [hunk 2]
```diff
@@ -2,7 +2,6 @@
 <?pde version="3.8"?>
 <target name="LSP4MP Target Platform">
 	<locations>
-		<location type="Target" uri="https://repo.eclipse.org/content/repositories/jdtls-snapshots/org/eclipse/jdt/ls/org.eclipse.jdt.ls.tp/1.16.0-SNAPSHOT/org.eclipse.jdt.ls.tp-1.16.0-20220915.034113-10.target"/>
```
Comment (rgrunber) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target:5]: Could we just comment this out and make the uri `https://raw.githubusercontent.com/eclipse/eclipse.jdt.ls/master/org.eclipse.jdt.ls.target/org.eclipse.jdt.ls.tp.target` ? For ease of local development ?

**Pull request merged**